Framework/Core/src/DataDescriptorMatcher.cxx

```diff
@@ -383,7 +390,11 @@ bool DataDescriptorMatcher::operator==(DataDescriptorMatcher const& other) const
   }
 
   if (mOp == Op::Just) {
-    return true;
+    return leftValue;
```

Comment:
I think when having two matchers with just left condition, the two should should be tested for equality and thus we need to return `leftValue`.

Actually, `leftResult` and `rightresult` would be better naming, shall we apply this?